superset-frontend/src/components/GenericLink/GenericLink.tsx

```diff
@@ -29,9 +30,9 @@ export const GenericLink = <S,>({
   ...rest
 }: React.PropsWithoutRef<LinkProps<S>> &
   React.RefAttributes<HTMLAnchorElement>) => {
-  if (typeof to === 'string' && /^https?:\/\//.test(to)) {
+  if (typeof to === 'string' && isUrlExternal(to)) {
     return (
-      <a data-test="external-link" href={to} {...rest}>
+      <a data-test="external-link" href={parseUrl(to)} {...rest}>
```

Comment:
Small suggestion for a follow-up: You could return the parsed value when it's valid and undefined or another thing when it's invalid. That way you would only need to apply the regex once by combining `isUrlExternal` with `parseUrl`.

Reply:
Good point! I'll do that in a separate PR